File: Creational/Prototype/Java/Example/Document.java
public class Document implements Prototype {
    public String title;
    public String content;
    public User author;

    public Document(String title, String content, User author) {
        this.title = title;
        this.content = content;
        this.author = author;
    }

    @Override
    public Document clone() {
        User author_clone = author.clone();
        return new Document(this.title, this.content, author_clone);
    }
}

File: Creational/Prototype/Python/Example/document.py
import copy

class Document:
    def __init__(self, title: str, content: str):
        self.title = title
        self.content = content

    def clone(self):
        # deep copy to ensure independence
        return copy.deepcopy(self)

    def __str__(self):
        return f"Document(title='{self.title}', content='{self.content}')"
